src/main/kotlin/org/rust/ide/inspections/RsSelfInStaticContextInspection.kt

```diff
@@ -0,0 +1,32 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.impl.mixin.isAssocFn
+import org.rust.lang.core.psi.util.parentOfType
+
+class RsSelfInStaticContextInspection : RsLocalInspectionTool() {
```

Comment:
All the Rust error detection is now performed in the [RsErrorAnnotator](https://github.com/intellij-rust/intellij-rust/blob/28b61a3e809a1034f6cfb5727a023d0f4a00e7c7/src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt) class. I think it would be more convenient to move it there.